File: Abubakr Khanooni/Arrays/DecompressRunLengthEncodedString.cpp
class Solution {
public:
    vector<int> decompressRLElist(vector<int>& nums) {
        vector<int>ams;
        for(int i=0;i<nums.size()-1;i+=2){
            int freq=nums[i];
            int val=nums[i+1];

            while(freq!=0){
                ams.push_back(val);
                freq--;
            }
        }
        return ams;
    }
};

File: Deepak Urs/Arrays/decompressRunLengthEncodedString.java
import java.util.*;
;
class Solution {
    public int[] decompressRLElist(int[] nums) {
        List<Integer> res = new ArrayList<>();
        int i = 0;
        int n = nums.length;

        while (i < n) {
            int frequency = nums[i];
            int value = nums[i + 1];
            for (int j = 0; j < frequency; j++) {
                res.add(value);
            }
            i += 2;
        }

        int[] resultArray = new int[res.size()];
        for (int k = 0; k < res.size(); k++) {
            resultArray[k] = res.get(k);
        }

        return resultArray;
        
    }
}